### ISSUE TITLE
Handle links in post titles

### ISSUE THREAD

**Issue by BrooklynMan** [mlemgroup/mlem#91]
App currently does not handle links embedded in post titles, and it's a feature in the web version.

<img width="661" alt="Screenshot 2023-06-26 at 7 14 00 PM" src="https://github.com/mormaer/Mlem/assets/5836168/a7c15a7e-d028-4a39-ab51-6641d393748b">

**Comment by d42ohpaz** [comment]
A few thoughts to aid in usability if links in titles are tapped: 

- Display a popover that asks the user to either open link or open post. 
- Long-press to open link. 
- Filter out the link and leave the display text; eg `[#foo](https://bar)` would become `#foo`
  - you could always make the navigation link active inside the post where it's less likely to be tapped accidentally

**Comment by pylapp** [comment]
I got the same issue this week.

In fact in _Lemmy_ we can, for a publication, create a new one with a title and a body, or create a new one with a title and an URL to share (making some suggestions for the title). I faced some troubles by just posting the hyperlink in the body, but users in the Lemmy place were a bit disturbed because they expected to have a dedicated rendering of that hyperlink, and not an hyperlink in the area which should contain plain text.

For example, given the following screen capture showing the Web UI, you see a user can define a title with an image and/or an URL and/or some text.

<img width="500" alt="webUI" src="https://github.com/user-attachments/assets/4ec56b6f-19e5-486c-9514-e2b7ebdfd29a">

But with the current version of Mlem, we don't have the specific feature for hyperlinks

<img width="500" alt="Mlem" src="https://github.com/user-attachments/assets/bae57965-b75e-4edc-a3cb-5d4ac29af3fb">

Thus, users will post things in Lemmy, with hyperlinks in the "text" and readers are disturbed because the hyperlink shuld be defined elswhere and implies a dedicated behavior.

[You can read this thread for further details](https://programming.dev/comment/12505261).

cc @EricBAndrews

**Comment by Sjmarf** [comment]
Yeah, we've noticed a lot of people getting confused about the Mlem v1 post creator. You _can_ actually add a link directly to the post by tapping the link button in the toolbar then tapping "Paste", but we realise this isn't a particularly obvious UI. We've improved the way it works in 2.0.
